### DISCUSSION TITLE
Long build time with Gradle and significant platform specific code

### DISCUSSION THREAD

**Discussion by jolindroth** [obytes/react-native-template-obytes#359] [Q&A] [answered]
Hi, great boilerplate guys and I really appreciate your efforts.

I noticed that, when following the instructions in your README, without any changes, and running pnpm android, the project begins building with gradle for 20 minutes. This is not how my experience has been with other expo projects, so I am wondering if this is expected from your end? 

If it is expected, it would be great to hear your motivations for going this route, and what the benefits are for including quite significant platform specific code in the android and ios folders.

Thanks again!

**Comment by yjose** [comment]
Hey @jolindroth  Normally, as we add some native dependencies, the build will take more time, but it shouldn't take that long. So, it is likely related to a library or a misconfiguration. can you give me more details on the device you are working on?

**Reply by jolindroth** [reply]
I'm running a windows workstation laptop. Subsequent builds just took a few minutes. 

Maybe a follow up question could be, if you're using CNG, does the native folders need to be included in the repo? I tried deleting the folders and gitignoring them and it seems to work just fine.

**Reply by yjose** [reply, marked as answer]
`if you're using CNG, does the native folders need to be included in the repo?`  Not really. You can add them to the .gitignore file if you want. From my experience working on a team, many forget to rerun the prebuild after pulling the latest updates so i always prefer pushing the iOS and Android folders.  But this is probably not the right solution. 😀

I don't have a strong opinion on that. Also, when you initiate a new project with the CLI, we usually delete the iOS and Android folders on the fly. They should be generated when you run `pnpm run ios`.
